server/service.go

```diff
@@ -737,26 +727,19 @@ func (m *BoostService) processDenebPayload(w http.ResponseWriter, req *http.Requ
 				}
 			}
 
-			// Lock before accessing the shared payload
-			mu.Lock()
-			defer mu.Unlock()
-
-			if requestCtx.Err() != nil { // request has been cancelled (or deadline exceeded)
-				return
-			}
-
-			// Received successful response. Now cancel other requests and return immediately
 			requestCtxCancel()
-			*result = *responsePayload
+			resultCh <- responsePayload
 			log.Info("received payload from relay")
 		}(relay)
 	}
 
 	// Wait for all requests to complete...
 	wg.Wait()
+	close(resultCh)
+	result := <-resultCh
```

Comment:
This is a strange pattern, but it should work as intended. We still need the wait group unfortunately. Normally, reading from a channel is a blocking operation, but if the channel is closed it's not blocking. It would be cool if we could figure out a way of handling this better, where we don't necessarily have to wait for each request to finish. As in, remove the wait group and proceed as soon as we receive the first request.

Reply:
I removed the wg and changed the pattern so that we only need one request to finish / or the timeout to trigger